src/app/home/home.component.ts

```diff
@@ -23,64 +34,104 @@ import { selectAllCountries } from '../store/country.selector';
     MatFormFieldModule,
     FormsModule,
     MatInputModule,
+    MatSlideToggleModule,
+    MatTableModule,
+    MatRadioModule,
   ],
   templateUrl: './home.component.html',
   styleUrls: ['./home.component.css'],
 })
 export class HomeComponent implements OnInit {
-  countries$: Observable<Country[]> = this.countryService.getAllCountries();
-  filteredCountries$: Observable<Country[]> = this.countries$;
+  disabled: boolean = false;
+  checked: boolean = false;
+  isTableView: boolean = false;
+  countries$: Observable<Country[]> = this.store.select(selectAllCountries);
+  filteredCountries$: Observable<Country[] | any> = this.countries$;
   searchTerm: string = '';
+  displayedColumns: string[] = [
+    'name',
+    'region',
+    'capital',
+    'continents',
+    'population',
+  ];
+  searchBy: 'name' | 'capital' = 'name';
+  routeSub: any;
+  isDialogOpen: boolean = false;
+  private dialogRef: MatDialogRef<CountryDetailComponent> | null = null;
 
   constructor(
     private store: Store,
-    private countryService: CountryService,
     private router: Router,
-    private dialog: MatDialog
+    private dialog: MatDialog,
+    private route: ActivatedRoute
   ) {}
 
   ngOnInit(): void {
-    // Load countries from store dispatch
+    // Load countries from the store
     this.store.dispatch(loadCountries());
-    // this.countries$ = this.store.select(selectAllCountries);
 
-    //  filtered countries
-    this.filteredCountries$ = this.countries$;
+    // Handle filtering logic
+    this.filteredCountries$ = this.store.select(
+      selectFilteredCountries(this.searchTerm, this.searchBy)
+    );
```

Comment:
why do we need to have a separate filteredCountries list.. we need to be able to update the countries list of the state with the results returned by the api. Then when the search term is removed get all request would be made.. Kindly change your implementation here.